odoo/addons/base/tests/test_expression.py

```diff
@@ -771,20 +774,25 @@ def test_invalid(self):
         """ verify that invalid expressions are refused, even for magic fields """
         Country = self.env['res.country']
 
-        with self.assertRaises(ValueError):
+        with self.assertRaises(ValueError) as cm:
             Country.search([('does_not_exist', '=', 'foo')])
+        self.assertEqual(cm.exception.args[0], "Invalid field res.country.does_not_exist in leaf ('does_not_exist', '=', 'foo')")
```

Comment:
You can use `self.assertRaisesRegex` to match the exception string representation.